server/game/cards/22-BtB/Gendry.js

```diff
@@ -0,0 +1,36 @@
+const DrawCard = require('../../drawcard.js');
+const {flatten} = require('../../../Array');
+
+class Gendry extends DrawCard {
+    setupCardAbilities(ability) {
+        this.persistentEffect({
+            condition: () => !this.controller.anyCardsInPlay(card => card.getType() === 'character' && card.isLoyal()),
+            match: this,
+            effect: ability.effects.addKeyword('Renown')
+        });
+        this.reaction({
+            when: {
+                onPhaseStarted: event => event.phase === 'challenge'
+            },
+            target: {
+                cardCondition: card => card.location === 'play area' && card.getType() === 'character' && !card.isLoyal()
+                                                        && card.attachments.some(attachment => attachment.hasTrait('Weapon'))
+            },
+            handler: context => {
+                let keywords = this.getKeywords();
+                this.untilEndOfPhase(ability => ({
+                    match: context.target,
+                    effect: [
+                        ability.effects.modifyStrength(2),
+                        flatten(keywords.map(keyword => ability.effects.addKeyword(keyword)))
```

Comment:
I used Vargo Hoat as a reference here, and can confirm that it definitely does work as intended on PT server (chosen character gets +2 STR & all of Gendry's keywords). I do understand your logic though, but not sure if it's worth an adjustment as its currently working fine, and is previously used on Vargo. :)

Reply:
Vargo Hoat has the `flatten` around the entire array, as I suggested above.

https://github.com/throneteki/throneteki/blob/f8ea72d5a7625b234f2dda60e7fcc8a9883ddc68/server/game/cards/18-FH/VargoHoat.js#L42

It appears to be working because of this line:
https://github.com/throneteki/throneteki/blob/f8ea72d5a7625b234f2dda60e7fcc8a9883ddc68/server/game/game.js#L262

I'd still recommend changing it.

Reply:
Fair enough! Adjusted in 25b8058ec012ab571992dc9ee71226e70b945c3e (along with message fix).